src/lib/utils/time.ts

```diff
@@ -0,0 +1,64 @@
+export const millisecondsPerSecond = 1000
+export const secondsPerMinute = 60
+export const minutesPerHour = 60
+export const hoursPerDay = 24
+export const daysPerWeek = 7
+
+export const millisecondsPerMinute = millisecondsPerSecond * secondsPerMinute
+export const millisecondsPerHour = millisecondsPerMinute * minutesPerHour
+export const millisecondsPerDay = millisecondsPerHour * hoursPerDay
+export const millisecondsPerWeek = millisecondsPerDay * daysPerWeek
+export const previousSevenDays = millisecondsPerWeek
+export const lastThirtyDays = millisecondsPerDay * 30
+
+export const currentYear = new Date().getFullYear()
+
+export function getTimeSince(date: Date) {
```

Comment:
This function actually returns the number of milliseconds since the supplied date, perhaps it should be called `millisecondsSince` or something similar?

Reply:
This function should possibly also return 0 for dates in the future, to avoid unexpected results?

Reply:
I made it return 0 if date is in the future, and also put a check in other functions that uses this millsecondsSince, like: while millisecondsSince != 0